### PR TITLE
[babel-preset] fix tests after RN bump

### PR DESCRIPTION
**Pull request by Simek**
# Why

Refs #30618
Fixes https://github.com/expo/expo/actions/runs/10092259422/job/27905552772?pr=30618

# How

Fix `babel-preset-expo` tests after latest RN bump.

# Test Plan

Snapshots are updated, all tests are passing.

### CODE REVIEW

**Review by gabrieldonadel — APPROVED** [review]
LGTM

File: packages/babel-preset-expo/src/__tests__/__snapshots__/compiler.test.ts.snap [hunk 1]
```diff
@@ -3,7 +3,7 @@
 exports[`allows destructuring in the catch block 1`] = `
 "Object.defineProperty(exports, "__esModule", { value: true });exports.App = App;var _jsxRuntime = require("react/jsx-runtime");function App() {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  var foo = function () {
+  var foo = () => {
```
Comment (EvanBacon) [inline packages/babel-preset-expo/src/__tests__/__snapshots__/compiler.test.ts.snap:6]: This is a pretty big change, did Hermes update and add support for this functionality?

Comment (EvanBacon) [reply packages/babel-preset-expo/src/__tests__/__snapshots__/compiler.test.ts.snap:6]: Not a ton of info here https://github.com/facebook/react-native/pull/44534

**Review by EvanBacon — APPROVED** [review]
My understanding is that this is a result of internal fixes at Meta. Would keep an eye on it though.

File: packages/babel-preset-expo/src/__tests__/__snapshots__/compiler.test.ts.snap (hunk 1 above)
Comment (Simek) [reply packages/babel-preset-expo/src/__tests__/__snapshots__/compiler.test.ts.snap:6]: Thanks for digging into root of this! 🙏

**Pull request merged**